lua/toggleterm/config.lua

```diff
@@ -17,10 +20,34 @@ local config = {
   shell = vim.o.shell,
   float_opts = {
     winblend = 0,
-    highlights = {
-      background = "Normal",
-      border = "Normal",
-    },
+  },
+}
+
+config.highlights = {
+  Normal = {
+    guibg = colors.shade_color(colors.get_hex("Normal", "bg"), constants.shading_amount),
+  },
+  NormalFloat = {
+    guibg = colors.get_hex("NormalFloat", "bg"),
+  },
+  FloatBorder = {
+    guifg = colors.get_hex("FloatBorder", "fg"),
+    guibg = colors.get_hex("FloatBorder", "bg"),
+  },
+  EndOfBuffer = {
+    guibg = colors.shade_color(colors.get_hex("Normal", "bg"), constants.shading_amount),
+  },
+  StatusLine = {
+    gui = "NONE",
+    guibg = colors.shade_color(colors.get_hex("Normal", "bg"), constants.shading_amount),
```

Comment:
The shade colour pattern is repeated a bunch, so you could create a small helper that does this automatically, e.g.
```lua
local function shade(color)
  return colors.shade(color, constants.shading_amount)
end
```